sdk/v1/client_test.go

```diff
@@ -679,3 +699,127 @@ func TestClient_GetTask(t *testing.T) {
 		})
 	})
 }
+
+func TestClient_GetTasks(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+	pathToCheck := "v1/jobs/883c81fd-726d-4ea3-9db8-7e7c781a01cc/tasks"
+
+	reqHeaders := client.Headers{
+		IfMatch:          "*",
+		ServiceAuthToken: "",
+	}
+
+	Convey("Given clienter.Do doesn't return an error", t, func() {
+		body, err := json.Marshal(exectedTasks)
+		if err != nil {
+			t.Errorf("failed to setup test data, error: %v", err)
+		}
+
+		httpClient := newMockHTTPClient(
+			&http.Response{
+				StatusCode: http.StatusCreated,
+				Body:       io.NopCloser(bytes.NewReader(body)),
+				Header: http.Header{
+					"Etag": []string{testETag},
+				},
+			},
+			nil)
+
+		searchReindexClient := newSearchReindexClient(t, httpClient)
+
+		Convey("When search-reindexClient.GetTasks is called", func() {
+			respHeaders, tasks, err := searchReindexClient.GetTasks(ctx, reqHeaders, testJobID)
+			So(err, ShouldBeNil)
+
+			Convey("Then the expected tasks list is returned", func() {
+				So(tasks.TaskList, ShouldHaveLength, 2)
+				So(tasks.Count, ShouldEqual, 2)
+				So(tasks.Limit, ShouldEqual, 5)
+				So(tasks.Offset, ShouldEqual, 0)
+				So(tasks.TotalCount, ShouldEqual, 20)
+				So(tasks.TaskList[0].JobID, ShouldEqual, "883c81fd-726d-4ea3-9db8-7e7c781a01cc")
+				So(tasks.TaskList[0].TaskName, ShouldEqual, "zebedee")
+				So(tasks.TaskList[0].NumberOfDocuments, ShouldEqual, 10)
+				So(tasks.TaskList[1].JobID, ShouldEqual, "883c81fd-726d-4ea3-9db8-7e7c781a01cc")
+				So(tasks.TaskList[1].TaskName, ShouldEqual, "dataset-api")
+				So(tasks.TaskList[1].NumberOfDocuments, ShouldEqual, 20)
+			})
+
+			Convey("And an ETag is returned", func() {
```

Comment:
Although an ETag is being returned by the test there is no ETag currently being returned, in the response header, if the GET tasks endpoint is actually called. So I think this part of the test should be removed as it is not currently doing this.

Reply:
correct, trello task for generating ETag in response header is - https://trello.com/c/flHAokOt/1152-search-reindex-api-ifmatch-and-etag-headers-for-get-jobs-jobid-tasks-endpoint

Reply:
I would right the client and unit test based on the swagger spec (assuming this is up to date)

In this case as Bhupendra says we will want to check the ETag is in the response header and not returned in the body. the data is mocked and so we can remove the ETag entry in the expectedTask variable when all the client methods have been updated.

See my next comment on removing duplicate assertions based on the above

Reply:
ETag entry removed with duplicate assertion.